### PR TITLE
fix(cmd): fix repository path handling for Init and Start commands

### PR DESCRIPTION
**Pull request by Wondertan**
Fixes bug introduced in #229, without the fix `start` does not work.

### CODE REVIEW

File: cmd/init.go [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 
 // Init constructs a CLI command to initialize Celestia Node of the given type 'tp'.
 // It is meant to be used a subcommand and also receive persistent flag name for repository path.
-func Init(repoName string, tp node.Type) *cobra.Command {
+func Init(repoFlagName string, tp node.Type) *cobra.Command {
 	if !tp.IsValid() {
 		panic("cmd: Init: invalid Node Type")
 	}
-	if len(repoName) == 0 {
+	if len(repoFlagName) == 0 {
```
Comment (renaynay) [inline cmd/init.go:17]: won't `len(repoFlagName)` never == 0 ? We're removing dev mode anyway which is the only thing that passes `""` to this function.

File: cmd/start.go [hunk 2]
```diff
@@ -35,6 +35,11 @@ func Start(repoName string, tp node.Type) *cobra.Command {
 		Args:         cobra.NoArgs,
 		SilenceUsage: true,
 		RunE: func(cmd *cobra.Command, args []string) error {
+			repoPath := cmd.Flag(repoFlagName).Value.String()
+			if repoPath == "" {
+				return fmt.Errorf("repository path must be speficied")
```
Comment (renaynay) [inline cmd/start.go:40]:
```suggestion
				return fmt.Errorf("repository path must be specified")
```

File: cmd/init.go (hunk 1 above)
Comment (Wondertan) [reply cmd/init.go:17]: Just a sanity check, as `Start` is exported and might be used outside. Does not hurt. Will revisit this when will delete Dev mode. Resolved?

Comment (Wondertan) [reply cmd/init.go:17]: @renaynay

**Review by liamsi — APPROVED** [review]

File: cmd/init.go (hunk 1 above)
Comment (liamsi) [reply cmd/init.go:17]: BTW, is there an issue to track removal of dev mode? 
Can this be closed then: https://github.com/celestiaorg/celestia-node/pull/176#issuecomment-985824614

**Comment by liamsi** [comment]
Is this tracked somewhere?
```
TestExtendedHeaderBroadcast
[...]
 event_test.go:100: 
        	Error Trace:	event_test.go:100
        	Error:      	Received unexpected error:
        	            	context deadline exceeded
        	Test:       	TestExtendedHeaderBroadcast
```

**Review by liamsi — APPROVED** [review]

File: cmd/init.go (hunk 1 above)
Comment (renaynay) [reply cmd/init.go:17]: Implicitly tracked here: https://github.com/celestiaorg/celestia-node/issues/250

Resolving.

**Review by renaynay — APPROVED** [review]

**Pull request merged**